processing_handler.go

```diff
@@ -264,6 +273,12 @@ func handleProcessing(reqID string, rw http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	if modifiedSince := r.Header.Get("If-Modified-Since"); len(modifiedSince) != 0 {
+		if _, ok := r.Header["If-None-Match"]; !ok {
```

Comment:
We should send both `If-Modified-Since` and `If-None-Match`. We don't know if the server supports ETags, so we should let the server decide which header to use. After all, we add `If-Modified-Since` support for the cases where the source server doesn't support ETags

Reply:
done!